src/CommunityToolkit.Aspire.Hosting.RavenDB/RavenDBBuilderExtensions.cs

```diff
@@ -0,0 +1,219 @@
+    /// <param name="databaseName">The name of the database to create/add. Defaults to the same name as the resource if not provided.</param>
+    /// <returns>A resource builder for the newly added RavenDB database resource.</returns>
+    /// <exception cref="DistributedApplicationException">Thrown when the connection string cannot be retrieved during configuration.</exception>
+    /// <exception cref="InvalidOperationException">Thrown when the connection string is unavailable.</exception>
+    public static IResourceBuilder<RavenDBDatabaseResource> AddDatabase(this IResourceBuilder<RavenDBServerResource> builder,
+        [ResourceName] string name,
+        string? databaseName = null)
+    {
+        ArgumentNullException.ThrowIfNull(builder);
+        ArgumentNullException.ThrowIfNull(name);
+
+        // Use the resource name as the database name if it's not provided
+        databaseName ??= name;
+
+        builder.Resource.AddDatabase(name, databaseName);
+        var databaseResource = new RavenDBDatabaseResource(name, databaseName, builder.Resource);
+
+        string? connectionString = null;
+
+        builder.ApplicationBuilder.Eventing.Subscribe<ConnectionStringAvailableEvent>(databaseResource, async (@event, ct) =>
+        {
+            connectionString = await databaseResource.ConnectionStringExpression.GetValueAsync(ct).ConfigureAwait(false);
+
+            if (connectionString == null)
+                throw new DistributedApplicationException($"ConnectionStringAvailableEvent was published for the '{databaseResource.Name}' resource but the connection string was null.");
+        });
+
+        var healthCheckKey = $"{name}_check";
+        builder.ApplicationBuilder.Services.AddHealthChecks()
+            .AddRavenDB(sp => connectionString ?? throw new InvalidOperationException("Connection string is unavailable"),
+                databaseName: databaseName,
+                name: healthCheckKey);
+
+        return builder.ApplicationBuilder
+            .AddResource(databaseResource);
+    }
+
+    /// <summary>
+    /// Adds a bind mount for the data folder to a RavenDB container resource.
+    /// </summary>
+    /// <param name="builder">The resource builder for the RavenDB server.</param>
+    /// <param name="source">The source directory on the host to mount into the container.</param>
+    /// <param name="isReadOnly">Indicates whether the bind mount should be read-only. Defaults to false.</param>
+    /// <returns>The <see cref="IResourceBuilder{T}"/> for the RavenDB server resource.</returns>
+    public static IResourceBuilder<RavenDBServerResource> WithDataBindMount(this IResourceBuilder<RavenDBServerResource> builder, string source, bool isReadOnly = false)
+    {
+        ArgumentNullException.ThrowIfNull(builder);
+        ArgumentNullException.ThrowIfNull(source);
+
+        return builder.WithBindMount(source, "/var/lib/ravendb/data", isReadOnly);
```

Comment:
shouldn't we set "/var/lib/ravendb/data" as the DataDir in env variables?
https://ravendb.net/docs/article-page/6.2/csharp/server/configuration/core-configuration#datadir

or its set by default?

Reply:
it is set by default